var documenterSearchIndex = {"docs":
[{"location":"man/intro/#Introduction-1","page":"Introduction","title":"Introduction","text":"","category":"section"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"SpeedDate is a method of estimating a divergence time between two DNA sequences. It was first implemented in the R package HybridCheck in order to date regions of introgression between homologous contigs of assembled genomes.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"This SpeedDate package implements the very same algorithm in Julia, making it faster than the original implementation. It is also built on top of packages in the BioJulia framework, including BioSequences.jl.","category":"page"},{"location":"man/intro/#How-does-the-estimation-method-work?-1","page":"Introduction","title":"How does the estimation method work?","text":"","category":"section"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"The divergence time is estimated using the number of mutations that have occurred between two aligned sequences.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"The calculation uses a strict molecular clock which assumes a constant substitution rate, both through time and across taxa.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Modelling the mutation accumulation process as a Bernoulli trial, the probability of observing k or fewer mutations between two sequences of length n can be given as:","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Pr(X le k) = sum_i=0^lfloor k rfloor binomni p^i (1 - p)^n-i","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Where p is the probability of observing that a single base position is a mutation between the two aligned sequences.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"The value of p depends on two key factors: the substitution rate and the divergence time.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"If you assume a molecular clock, whereby two DNA sequences are both accumulating mutations at a rate mu for t generations, then you may define p = 2mu t.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Using these assumptions, the SpeedDate method finds the root of the following formula for Pr(X le k) = 005, 05, and 095, and then divides the three answers by twice the assumed substitution rate.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"f(n k 2mu t Pr(X le k) = left( sum_i=0^lfloor k rfloor binomni 2mu t^i (1 - 2mu t)^n-i   right) - Pr(X le k)","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"This results in an upper, middle, and lower estimate of the coalescence time t of the two sequences (expressed as the number of generations).","category":"page"},{"location":"man/intro/#Loading-SpeedDate-1","page":"Introduction","title":"Loading SpeedDate","text":"","category":"section"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"To begin using SpeedDate, fire up a Julia session enter the following into the prompt:","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"using SpeedDate","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"note: Note\nThis assumes you have installed SpeedDate already. See the quick-start section of the homepage to see how to install SpeedDate.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Once that's done, you're ok to continue and do one of the analyses described in the rest of this manual.","category":"page"},{"location":"man/dating/#Dating-Sequences-1","page":"Dating Sequences","title":"Dating Sequences","text":"","category":"section"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"You use SpeedDate in a very simple 2 step manner. First, count the mutations between two sequences, and then you plug that information, plus a mutation rate into the dating function.","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"You can use the basic functions, which work on a pair of sequences, plus other BioJulia utilities to run any analysis of any combination of sequences you may like. But there are also some extra methods of the basic functions provided for convenience e.g. running sating between sequences in a sliding window, or doing a pairwise analysis. The dating algorithm is the same in all cases, but you can run it for two sequences, or pairwise between many sequences, or using a sliding window.","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"Let's see some example use cases:","category":"page"},{"location":"man/dating/#Dating-a-pair-of-homologous-sequences-1","page":"Dating Sequences","title":"Dating a pair of homologous sequences","text":"","category":"section"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"To estimate the divergence time of a single pair of homologous sequences, first use the count_mutations function, to get the number of mutations that are observed between the two sequences. Then use the estimate_time function on the result, providing a value for the assumed substitution rate μ. E.g.","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"m_count = count_mutations(seqa, seqb)\nt_estimate = estimate_time(m_count, 10e-7) # μ = 10e-7","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"estimate_time returns a DatingEstimate struct that represents a CI  for the divergence time estimate by containing values for the 5%, 50% and 95% estimates.","category":"page"},{"location":"man/dating/#Pairwise-dating-of-a-set-of-homologous-sequences-1","page":"Dating Sequences","title":"Pairwise dating of a set of homologous sequences","text":"","category":"section"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"If you have a set of homologous sequences - say genes - and you want to estimate the divergence time between each pair of sequences, SpeedDate allows you to do this, giving you a symmetrical matrix as a result, where each cell is the result for a pair of sequences.","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"First you use count_mutions on a vector of sequences, which gives you a matrix of mutation counts. Then you can use estimate_time to get a matrix of time estimates. E.g.","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"count_matrix = count_mutations([seqa, seqb, seqc, seqd])\nt_matrix = estimate_time(m_count, 10e-7) # μ = 10e-7","category":"page"},{"location":"man/dating/#Dating-a-pair-of-homologous-sequences-over-a-sliding-window-1","page":"Dating Sequences","title":"Dating a pair of homologous sequences over a sliding window","text":"","category":"section"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"To estimate the divergence time of a single pair of homologous sequences, using a sliding window, first use the count_mutations function, providing the two sequences, as well as a value to use as the size (in base-pairs) of the sliding window. The result can then be used with the estimate_time function. E.g.","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"window_counts = count_mutations(seqa, seqb, 50) # 50bp sliding window.\nwindow_estimate = estimate_time(window_counts, 10e-7) # μ = 10e-7","category":"page"},{"location":"man/loading/#Loading-Data-1","page":"Loading Data","title":"Loading Data","text":"","category":"section"},{"location":"man/loading/#","page":"Loading Data","title":"Loading Data","text":"First things first you need to load your sequence data in order to use it in SpeedDate.","category":"page"},{"location":"man/loading/#","page":"Loading Data","title":"Loading Data","text":"SpeedDate currently supports the FASTA format for sequences, and assumed the sequences have been aligned with some kind of multiple sequence alignment program, such as Clustal. As a result all of the sequences typically should be of the same length as the alignment process adds gap characters '-' to the sequences.","category":"page"},{"location":"man/loading/#","page":"Loading Data","title":"Loading Data","text":"To read in a FASTA file use the read_fasta function:","category":"page"},{"location":"man/loading/#","page":"Loading Data","title":"Loading Data","text":"my_seqs = read_fasta(\"my_sequences.fasta\")","category":"page"},{"location":"#SpeedDate-1","page":"Home","title":"SpeedDate","text":"","category":"section"},{"location":"#","page":"Home","title":"Home","text":"(Image: Latest release) (Image: MIT license) (Image: Stable documentation) (Image: Dev documentation) (Image: Pkg Status)","category":"page"},{"location":"#Description-1","page":"Home","title":"Description","text":"","category":"section"},{"location":"#","page":"Home","title":"Home","text":"A utility for rapidly estimating coalescence times between sequences.","category":"page"},{"location":"#Quick-start-installation-1","page":"Home","title":"Quick-start installation","text":"","category":"section"},{"location":"#","page":"Home","title":"Home","text":"Fire up a julia interactive window.","category":"page"},{"location":"#","page":"Home","title":"Home","text":"In the window press the ']' key.","category":"page"},{"location":"#","page":"Home","title":"Home","text":"You should observe the prompt change from julia> to pkg>.","category":"page"},{"location":"#","page":"Home","title":"Home","text":"Type in add SpeedDate and hit enter.","category":"page"},{"location":"#","page":"Home","title":"Home","text":"Press backspace and you should see the prompt change back to julia>.","category":"page"},{"location":"#","page":"Home","title":"Home","text":"That's it!","category":"page"}]
}
