var documenterSearchIndex = {"docs":
[{"location":"man/gtkgui/#Running-SpeedDate-with-a-GTK-based-Graphical-User-Interface-1","page":"Running SpeedDate with a GTK based Graphical User Interface","title":"Running SpeedDate with a GTK based Graphical User Interface","text":"","category":"section"},{"location":"man/gtkgui/#","page":"Running SpeedDate with a GTK based Graphical User Interface","title":"Running SpeedDate with a GTK based Graphical User Interface","text":"If you use the interactive command instead of the compute command, you will be presented with a GTK based graphical user interface.","category":"page"},{"location":"man/intro/#Introduction-1","page":"Introduction","title":"Introduction","text":"","category":"section"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"SpeedDate is a method of estimating a divergence time between two DNA sequences. It was first implemented in the R package HybridCheck in order to date regions of introgression between homologous contigs of assembled genomes.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"This SpeedDate package implements the very same algorithm in Julia, making it faster than the original implementation. It is also built on top of packages in the BioJulia framework, including BioSequences.jl.","category":"page"},{"location":"man/intro/#How-does-the-estimation-method-work?-1","page":"Introduction","title":"How does the estimation method work?","text":"","category":"section"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"The divergence time is estimated using the number of mutations that have occurred between two aligned sequences.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"The calculation uses a strict molecular clock which assumes a constant substitution rate, both through time and across taxa.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Modelling the mutation accumulation process as a Bernoulli trial, the probability of observing k or fewer mutations between two sequences of length n can be given as:","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Pr(X le k) = sum_i=0^lfloor k rfloor binomni p^i (1 - p)^n-i","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Where p is the probability of observing that a single base position is a mutation between the two aligned sequences.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"The value of p depends on two key factors: the substitution rate and the divergence time.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"If you assume a molecular clock, whereby two DNA sequences are both accumulating mutations at a rate mu for t generations, then you may define p = 2mu t.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Using these assumptions, the SpeedDate method finds the root of the following formula for Pr(X le k) = 005, 05, and 095, and then divides the three answers by twice the assumed substitution rate.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"f(n k 2mu t Pr(X le k) = left( sum_i=0^lfloor k rfloor binomni 2mu t^i (1 - 2mu t)^n-i   right) - Pr(X le k)","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"This results in an upper, middle, and lower estimate of the coalescence time t of the two sequences (expressed as the number of generations).","category":"page"},{"location":"man/intro/#Loading-SpeedDate-1","page":"Introduction","title":"Loading SpeedDate","text":"","category":"section"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"To begin using SpeedDate, fire up a Julia session enter the following into the prompt:","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"using SpeedDate","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"note: Note\nThis assumes you have installed SpeedDate already. See the quick-start section of the homepage to see how to install SpeedDate.","category":"page"},{"location":"man/intro/#","page":"Introduction","title":"Introduction","text":"Once that's done, you're ok to continue and do one of the analyses described in the rest of this manual.","category":"page"},{"location":"man/dating/#Dating-Sequences-1","page":"Dating Sequences","title":"Dating Sequences","text":"","category":"section"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"SpeedDate can perform its dating estimate in a few different ways. The algorithm is the same in all cases, but you can run it for two sequences, or pairwise between many sequences, or using a sliding window.","category":"page"},{"location":"man/dating/#Dating-a-pair-of-homologous-sequences-1","page":"Dating Sequences","title":"Dating a pair of homologous sequences","text":"","category":"section"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"To estimate the divergence time of a single pair of homologous sequences, first use the count_mutations function, to get the number of mutations that are observed between the two sequences. Then use the estimate_time function on the result, providing a value for the assumed substitution rate μ. E.g.","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"m_count = count_mutations(seqa, seqb)\nt_estimate = estimate_time(m_count, 10e-7) # μ = 10e-7","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"estimate_time returns a DatingEstimate struct that represents a CI  for the divergence time estimate by containing values for the 5%, 50% and 95% estimates.","category":"page"},{"location":"man/dating/#Pairwise-dating-of-a-set-of-homologous-sequences-1","page":"Dating Sequences","title":"Pairwise dating of a set of homologous sequences","text":"","category":"section"},{"location":"man/dating/#Dating-a-pair-of-homologous-sequences-over-a-sliding-window-1","page":"Dating Sequences","title":"Dating a pair of homologous sequences over a sliding window","text":"","category":"section"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"To estimate the divergence time of a single pair of homologous sequences, using a sliding window, first use the count_mutations function, providing the two sequences, as well as a value to use as the size (in base-pairs) of the sliding window. The result can then be used with the estimate_time function. E.g.","category":"page"},{"location":"man/dating/#","page":"Dating Sequences","title":"Dating Sequences","text":"window_counts = count_mutations(seqa, seqb, 50) # 50bp sliding window.\nwindow_estimate = estimate_time(window_counts, 10e-7) # μ = 10e-7","category":"page"},{"location":"man/injulia/#Using-SpeedDate-from-a-julia-script-or-session-1","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"","category":"section"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"You can execute the SpeedDate program from within a julia console session or script.","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"To do this, create a dictionary of arguments and then give that to the SpeedDate.compute function.","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"using SpeedDate\n\noptions = Dict(\"model\" => \"jc69\",\n               \"mutation_rate\" => 10e-9,\n               \"method\" => \"default\",\n               \"scan\" => false,\n               \"onlydist\" => false,\n               \"file\" => \"contig1_ac.fas\",\n               \"outfile\" => \"test\")\n\nSpeedDate.compute(options)","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"The dictionary should have the following fields:","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"\"model\": Either \"jc69\" or \"k80\".\n\"mutation_rate\": A number representing the mutation rate, e.g. 10e-9.\n\"file\": A string that is the filename e.g. \"~/Desktop/myseqs.fasta\".\n\"outfile\": A string that is the basename of all output files.\n\"method\": Either \"default\" or \"simple\", I recommend leaving as \"default\".\n\"onlydist\": Set to true of false depending on whether or not you only want the distances to be computed.\n\"scan\": false.","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"If you want to do the sliding window scan, the following should also be added to the dictionary:","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"\"scan\": true\n\"size\": A number representing the sliding window size.\n\"step\": A number representing the sliding window step.","category":"page"},{"location":"man/injulia/#Plotting-SpeedDate-results-from-a-julia-script-or-session-1","page":"Using SpeedDate from a julia script or session","title":"Plotting SpeedDate results from a julia script or session","text":"","category":"section"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"You can plot the results of the SpeedDate program from within a julia console session or script.","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"To do this, create a dictionary of arguments and then give that to the SpeedDate.visualize function.","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"using SpeedDate\n\noptions = Dict(\"width\" => 12.0,\n               \"height\" => 8.0,\n               \"units\" => \"cm\",\n               \"backend\" => \"svg\",\n               \"reference\" => \"default\",\n               \"table\" => false,\n               \"sortsim\" => false,\n               \"inputfile\" => \"myresults.csv\",\n               \"outputfile\" => \"myplot.svg\")\n\nSpeedDate.visualize(options)\n","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"The arguments dictionary should have the following fields:","category":"page"},{"location":"man/injulia/#","page":"Using SpeedDate from a julia script or session","title":"Using SpeedDate from a julia script or session","text":"\"width\": Width of the plot.\n\"height\": Height of the plot.\n\"units\": Units for width and height of the plot (\"cm\", \"mm\" or \"inch\").\n\"backend\": The backend used to produce the plot.\n\"reference\": The name of the DNA sequence to use as a reference when plotting a windowed analysis.\n\"table\": Set to true to write out the table that was generated for plotting - only really useful for debugging.\n\"sortsim\": Set to true to sort the rows of the output table by average sequence similarity if you are plotting a windowed analysis. \n\"inputfile\": The pathname of the SpeedDate results file you want to plot.\n\"outputfile\": A name for the output plot.","category":"page"},{"location":"man/loading/#Loading-Data-1","page":"Loading Data","title":"Loading Data","text":"","category":"section"},{"location":"man/loading/#","page":"Loading Data","title":"Loading Data","text":"First things first you need to load your sequence data in order to use it in SpeedDate.","category":"page"},{"location":"man/loading/#","page":"Loading Data","title":"Loading Data","text":"SpeedDate currently supports the FASTA format for sequences, and assumed the sequences have been aligned with some kind of multiple sequence alignment program, such as Clustal. As a result all of the sequences typically should be of the same length as the alignment process adds gap characters '-' to the sequences.","category":"page"},{"location":"man/loading/#","page":"Loading Data","title":"Loading Data","text":"To read in a FASTA file use the read_fasta function:","category":"page"},{"location":"man/loading/#","page":"Loading Data","title":"Loading Data","text":"my_seqs = read_fasta(\"my_sequences.fasta\")","category":"page"},{"location":"#","page":"Home","title":"Home","text":"Hello","category":"page"}]
}
